### PR TITLE
Gcf composer trigger

### PR DESCRIPTION
**Pull request by leahecole**
This code will be part of the [Triggering DAGs (workflows)](https://cloud.google.com/composer/docs/how-to/using/triggering-with-gcf) tutorial - ping me for internal docs CL if you'd like

### CODE REVIEW

File: functions/composer/composer_storage_trigger.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    # Navigate to your webserver's login page and get this from the URL
+    # Or use the script found at
+    # https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/composer/rest/get_client_id.py
+    client_id = 'YOUR-CLIENT-ID'
+    # This should be part of your webserver's URL:
+    # {tenant-project-id}.appspot.com
+    webserver_id = 'YOUR-TENANT-PROJECT'
+    # The name of the DAG you wish to trigger
+    dag_name = 'composer_sample_trigger_response_dag'
+    webserver_url = 'https://' + webserver_id + \
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]: Nit: Prefer parentheses for multi-line statements. 

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces.

Also, the `+` operator should start the line.

https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

```suggestion
    webserver_url = (
        'https://'
        + webserver_id
        + '.appspot.com/api/experimental/dags/'
        + dag_name
        + '/dag_runs'
    )
```

File: functions/composer/composer_storage_trigger.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        '.appspot.com/api/experimental/dags/' + dag_name + '/dag_runs'
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]:
```suggestion
```

File: functions/composer/composer_storage_trigger.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import requests
+import requests_toolbelt.adapters.appengine
+IAM_SCOPE = 'https://www.googleapis.com/auth/iam'
+OAUTH_TOKEN_URI = 'https://www.googleapis.com/oauth2/v4/token'
+
+
+def trigger_dag(data, context=None):
+    """Makes a POST request to the Composer DAG Trigger API
+
+    When called via GCF, data and context are Background function parameters.
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]: Nit: Use "Google Cloud Functions (GCF)" the first time that the product name appears in a file.

File: functions/composer/composer_storage_trigger.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    For more info, refer to
+    https://cloud.google.com/functions/docs/writing/background#functions_background_parameters-python
+
+    When called via Python, a context object is not needed,
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]: Nit: Prefer active tense. Also, I'd like to know what keys are expected in the data argument.

```suggestion
    To call this function from a Python script, omit the ``context`` argument
    and pass in a non-null value for the ``data`` argument.
    """
```

File: functions/composer/composer_storage_trigger.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    and a non-null data object must be present."""
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]:
```suggestion
```

File: functions/composer/composer_storage_trigger.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+import google.auth
+import google.auth.app_engine
+import google.auth.compute_engine.credentials
+import google.auth.iam
+from google.auth.transport.requests import Request
+import google.oauth2.credentials
+import google.oauth2.service_account
+import requests
+import requests_toolbelt.adapters.appengine
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]: Is this necessary when using GCF? I thought the titanium runtime didn't need any bootstrapping?

File: functions/composer/composer_storage_trigger.py (hunk 1 above)
Comment (leahecole) [reply functions/composer/composer_storage_trigger.py]: Ooo thank you for the references. I'm not as good with python style as I am Node!

File: functions/composer/composer_storage_trigger.py (hunk 6 above)
Comment (leahecole) [reply functions/composer/composer_storage_trigger.py]: It is not! I removed all appengine related auth things to simplify

File: functions/composer/composer_storage_trigger.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+# [START composer_trigger]
+
+import google.auth
+import google.auth.compute_engine.credentials
+import google.auth.iam
+from google.auth.transport.requests import Request
+import google.oauth2.credentials
+import google.oauth2.service_account
+import requests
+IAM_SCOPE = 'https://www.googleapis.com/auth/iam'
```
Comment (tswast) [inline functions/composer/composer_storage_trigger.py]: Nit: Separate imports from code by 2 blank lines.

```suggestion


IAM_SCOPE = 'https://www.googleapis.com/auth/iam'
```

**Pull request merged**